Added device 009-128 (air conditioner)

psycho202 opened the pull request:
Added my model of AC. Seems similar to existing model 09-100, so copied but adjusted presets for my own use case. 




Further improvements to data_dictionary for this model and possibly others: 
- property f_votage corresponds to grid voltage, as measured by the indoor unit. 
Couldn't get this to work to add it as a property with the correct sensor device class. 

- property f_cool_qvalue corresponds to the BTU/hr of the unit, how much heat it's removing at that moment. Cool data to know how "hard" the aircon is working at a certain point in time.

- property t_fan_speed_s remains equal to t_fan_speed, so this property can probably be disabled by default?

psycho202 commented:
![image](https://github.com/user-attachments/assets/99131f40-1a95-4bfd-a052-b6edbd42b70d)
Example of f_cool_qvalue values in HA for this model

oyvindwe commented:
Thank you for your contribution! ❤️ 

> property f_votage corresponds to grid voltage, as measured by the indoor unit.
> Couldn't get this to work to add it as a property with the correct sensor device class.
`f_votage` or `f_voltage`? Spelling errors seem to be quite common in this API. Probably not very useful, but did you try this:
```
- property: f_votage
  sensor:
    device_class: voltage
    unit: V
```

> property f_cool_qvalue corresponds to the BTU/hr of the unit, how much heat it's removing at that moment. Cool data to know how "hard" the aircon is working at a certain point in time.

I don't see any matching device class, but this should work:
```
- property: f_cool_qvalue
  sensor:
    unit: BTU/hr
    state_class: measurement
```

> property t_fan_speed_s remains equal to t_fan_speed, so this property can probably be disabled by default?

Unmapped properties are hidden by default. Currently no support to disable properties completely, but that may be a nice feature to add.

psycho202 commented:
Thanks @oyvindwe, tested and added the properties!

f_votage indeed is a typo in the API. Also added the f_heat_qvalue, some icons and their text mapping to the strings.json and en.json files.

oyvindwe commented on custom_components/connectlife/strings.json:
```diff
@@ -621,6 +621,15 @@
         "name": "Number of zones"
+      },
+      "f_heat_qvalue": {
+        "name": "BTU/Hr Heating"
```
```
        "name": "Heating"
```

oyvindwe commented on custom_components/connectlife/strings.json:
```diff
@@ -621,6 +621,15 @@
+        "name": "BTU/Hr Heating"
+      },
+      "f_cool_qvalue": {
+        "name": "BTU/Hr Cooling"
```
```
        "name": "Cooling"
```

oyvindwe commented on custom_components/connectlife/strings.json:
```diff
@@ -621,6 +621,15 @@
+        "name": "BTU/Hr Cooling"
+      },
+      "f_votage": {
+        "name": "Measured Grid Voltage"
```
```
        "name": "Measured grid voltage"
```

oyvindwe commented on custom_components/connectlife/translations/en.json:
```diff
@@ -621,6 +621,15 @@
                 "name": "Number of zones"
+            },
+            "f_heat_qvalue": {
+              "name": "BTU/Hr Heating"
```
```
        "name": "Heating"
```

oyvindwe commented on custom_components/connectlife/translations/en.json:
```diff
@@ -621,6 +621,15 @@
+              "name": "BTU/Hr Heating"
+            },
+            "f_cool_qvalue": {
+              "name": "BTU/Hr Cooling"
```
```
        "name": "Cooling"
```

oyvindwe commented on custom_components/connectlife/translations/en.json:
```diff
@@ -621,6 +621,15 @@
+              "name": "BTU/Hr Cooling"
+            },
+            "f_votage": {
+              "name": "Measured Grid Voltage"
```
```
        "name": "Measured grid voltage"
```

oyvindwe commented on custom_components/connectlife/data_dictionaries/009-128.yaml:
```diff
@@ -0,0 +1,92 @@
+  - property: f_cool_qvalue
+    icon: mdi:snowflake
+    sensor:
+      unit: BTU/hr
```
```
      unit: BTU/h
```

oyvindwe commented on custom_components/connectlife/data_dictionaries/009-128.yaml:
```diff
@@ -0,0 +1,92 @@
+  - property: f_heat_qvalue
+    icon: mdi:weather-sunny
+    sensor:
+      unit: BTU/hr
```
```
      unit: BTU/h
```

oyvindwe reviewed (changes requested):
Looks good! Only thing is that it should not be necessary to include the unit in the property strings.

Preferred SI abbreviation for hours is h (apologies for not stating this in previous comment).

oyvindwe reviewed (approved):
I committed the requested changes.

oyvindwe commented:
Tip for future PRs: Make a _branch_ in your forks before committing changes, so you can keep main in sync with upstream repo.

psycho202 commented:
> Tip for future PRs: Make a _branch_ in your forks before committing changes, so you can keep `main` in sync with upstream repo.

Thanks for the tip! I'm not a developer by trade, so I'm not very used to working with github :)
Trying my best though!
